src/test/java/com/google/sps/EditServletTest.java

```diff
@@ -0,0 +1,520 @@
+import com.google.sps.servlets.AuthServlet;
+import com.google.sps.servlets.EditServlet;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({AuthServlet.class, DatastoreServiceFactory.class})
+public final class EditServletTest {
+
+  private EditServlet editServlet;
+
+  // Add constants for testing User class.
+  public static final String EMAIL = "testemail@gmail.com";
+
+  // Constants to represent different TripDay attributes.
+  private static final String INPUT_DESTINATION = 
+    "Space Needle, Broad Street, Seattle, WA, USA";
+  private static final String INPUT_DESTINATION_2 = 
+    "Woodland Park Zoo, Phinney Avenue North, Seattle, WA, USA";
+  private static final String INPUT_DATE = "2020-08-22";
+  private static final String INPUT_DATE_2 = "2020-08-23";
+
+  // Constants to represent different Trip attributes.
+  private static final String TRIP_NAME = "Trip to California";
+  private static final String DESTINATION_NAME = "California";
+  private static final String IMAGE_SRC =
+    "https://lh3.googleusercontent.com/p/AF1QipM7tbCZOj_5SOft9cYgI7un3bmieieqvdYkCPT5=s1600-w400";
+  private static final String TRIP_START_DATE = "2020-08-22";
+  private static final String TRIP_END_DATE = "2020-08-23";
+
+  // Constants to represent the different Event Attributes.
+  private static final String SPACE_NEEDLE = "Space Needle";
+  private static final String SPACE_NEEDLE_ADDRESS = 
+    "Space Needle, Broad Street, Seattle, WA, USA";
+  private static final LocalDateTime SPACE_NEEDLE_START_TIME = 
+      LocalDateTime.of(LocalDate.parse("2020-08-22"), LocalTime.of(11, 30));
+  private static final String SPACE_NEEDLE_PLACE_ID = "1234";
+  private static final String WOODLAND = "Woodland Park Zoo";
+  private static final String WOODLAND_ADDRESS = 
+    "Woodland Park Zoo, Phinney Avenue North, Seattle, WA, USA";
+  private static final LocalDateTime WOODLAND_START_TIME = 
+      LocalDateTime.of(LocalDate.parse("2020-08-22"), LocalTime.of(10, 00));
+      private static final String WOODLAND_PLACE_ID = "5678";
+  private static final int HALF_HOUR = 30;
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void initEditServlet() {
+    editServlet = new EditServlet();
```

Comment:
Do these tests work if you move this line to setUp()? It's stylistically fine to do all the initialization in a single `@Before` method.

Reply:
Ok, that makes sense -- fixed!